FeedStoreChallenge/CoreDataFeedStore.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  CoreDataFeedStore.swift
+//  FeedStoreChallenge
+//
+//  Created by Mohammed Al Waili on 01/03/2021.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+final public class CoreDataFeedStore: FeedStore {
+	
+	private let persistentContainer: NSPersistentContainer
+	private let context: NSManagedObjectContext
+	
+	public init(storeURL: URL, bundle: Bundle = .main) throws {
+		self.persistentContainer = try NSPersistentContainer.load(modelName: "LocalFeedImageModel", url: storeURL, in: bundle)
+		self.context = self.persistentContainer.newBackgroundContext()
+	}
+	
+	public func deleteCachedFeed(completion: @escaping DeletionCompletion) {
+		perform { context in
+			do {
+				let request = NSFetchRequest<CacheEntity>(entityName: "CacheEntity")
```

Comment:
You can use the static helper method instead:

```suggestion
				let request = CacheEntity.fetchRequest()
```